testsuite/tests/singlecluster/gateway/reconciliation/change_targetref/conftest.py

```diff
@@ -0,0 +1,95 @@
+"""
+Test for changing targetRef field in policies
+"""
+
+import pytest
+
+from testsuite.gateway import GatewayRoute, GatewayListener, Hostname, Exposer
+from testsuite.gateway.gateway_api.gateway import KuadrantGateway
+from testsuite.gateway.gateway_api.hostname import DNSPolicyExposer
+from testsuite.gateway.gateway_api.route import HTTPRoute
+from testsuite.kuadrant.policy.dns import DNSPolicy
+
+pytestmark = [pytest.mark.kuadrant_only, pytest.mark.dnspolicy]
+
+
+@pytest.fixture(scope="module")
+def exposer2(request, cluster) -> Exposer:
+    """Second DNSPolicyExposer setup for Gateway B"""
+    exposer = DNSPolicyExposer(cluster)
+    request.addfinalizer(exposer.delete)
+    exposer.commit()
+    return exposer
+
+
+@pytest.fixture(scope="module")
+def base_domain2(exposer2):
+    """Returns preconfigured base domain for the second Gateway"""
+    return exposer2.base_domain
+
+
+@pytest.fixture(scope="module")
+def wildcard_domain2(base_domain2):
+    """Wildcard domain for Gateway B"""
+    return f"*.{base_domain2}"
+
+
+@pytest.fixture(scope="module")
+def gateway(request, cluster, blame, wildcard_domain, module_label):
+    """Create and configure Gateway A"""
+    gw = KuadrantGateway.create_instance(cluster, blame("gw"), {"app": module_label})
+    gw.add_listener(GatewayListener(hostname=wildcard_domain))
+    request.addfinalizer(gw.delete)
+    gw.commit()
+    gw.wait_for_ready()
+    return gw
+
+
+@pytest.fixture(scope="module")
+def gateway_b(request, cluster, blame, wildcard_domain2, module_label):
+    """Create and configure Gateway B"""
+    gw = KuadrantGateway.create_instance(cluster, blame("gw-b"), {"app": module_label})
+    gw.add_listener(GatewayListener(hostname=wildcard_domain2))
+    request.addfinalizer(gw.delete)
+    gw.commit()
+    gw.wait_for_ready()
+    return gw
+
+
+@pytest.fixture(scope="module")
+def hostname_b(gateway_b, exposer2, blame) -> Hostname:
+    """Expose Hostname for Gateway B"""
+    hostname = exposer2.expose_hostname(blame("hostname-b"), gateway_b)
+    return hostname
+
+
+@pytest.fixture(scope="module")
+def route_b(request, gateway_b, blame, hostname_b, module_label, backend) -> GatewayRoute:
+    """Create and configure Route B"""
+    route = HTTPRoute.create_instance(gateway_b.cluster, blame("route-b"), gateway_b, {"app": module_label})
+    route.add_hostname(hostname_b.hostname)
+    route.add_backend(backend)
+    request.addfinalizer(route.delete)
+    route.commit()
+    route.wait_for_ready()
+    return route
+
+
+@pytest.fixture(scope="module")
+def client_b(route_b, hostname_b):  # pylint: disable=unused-argument
+    """Returns httpx client for Gateway B"""
+    client = hostname_b.client()
+    yield client
+    client.close()
+
+
+@pytest.fixture(scope="module")
+def dns_policy_b(blame, gateway_b, module_label, dns_provider_secret, request):
```

Comment:
we have some naming conventions for the fixutres in the testsuite, we are usually naming second resources with the number suffix, e.g. route, route2, gateway, gateway2, etc.

same with the `rate_limit_policy` fixture, which is just `rate_limit` in all cases